packages/caver-transaction/src/transactionTypes/abstractTransaction.js

```diff
@@ -0,0 +1,336 @@
+        let fillVariables = false
+        if (this.signatures.length === 0) fillVariables = true
+
+        for (const encoded of rlpEncodedTxs) {
+            const type = typeDetectionFromRLPEncoding(encoded)
+            if (this.type !== type) throw new Error(`Transaction type mismatch: Different transactions cannot acquire signatures.`)
+
+            const decoded = this.constructor.decode(encoded)
+
+            // Signatures can only be combined for the same transaction.
+            // Therefore, compare whether the decoded transaction is the same as this.
+            for (const k in decoded) {
+                if (k === '_signatures' || k === '_feePayerSignatures') continue
+                if (this[k] === undefined && fillVariables) this[k] = decoded[k]
+
+                const differentTxError = `Transactions containing different information cannot be combined.`
+
+                // Compare with RLP-encoded accountKey string, because 'account' is object.
+                if (k === '_account') {
+                    if (this[k].getRLPEncodingAccountKey() !== decoded[k].getRLPEncodingAccountKey()) throw new Error(differentTxError)
+                    continue
+                }
+
+                if (this[k] !== decoded[k]) throw new Error(differentTxError)
+            }
+
+            this.appendSignatures(decoded.signatures)
+        }
+
+        return this.getRLPEncoding()
+    }
+
+    /**
+     * Returns RawTransaction(RLP-encoded transaction string)
+     *
+     * @return {string}
+     */
+    getRawTransaction() {
+        return this.getRLPEncoding()
+    }
+
+    /**
+     * Returns a hash string of transaction
+     *
+     * @return {string}
+     */
+    getTransactionHash() {
+        return Hash.keccak256(this.getRLPEncoding())
+    }
+
+    /**
+     * Returns a senderTxHash of transaction
+     *
+     * @return {string}
+     */
+    getSenderTxHash() {
+        const rlpEncoded = this.getRLPEncoding()
+        return Hash.keccak256(rlpEncoded)
+    }
+
+    /**
+     * Returns an RLP-encoded transaction string for signing
+     *
+     * @return {string}
+     */
+    getRLPEncodingForSigning() {
+        if (this.gasPrice === undefined) throw new Error(`gasPrice is undefined. Use 'transaction.fillTransaction' to fill values.`)
+        if (this.nonce === undefined) throw new Error(`nonce is undefined. Use 'transaction.fillTransaction' to fill values.`)
+        if (this.chainId === undefined) throw new Error(`chainId is undefined. Use 'transaction.fillTransaction' to fill values.`)
+
+        return RLP.encode([this.getCommonRLPForSigning(), Bytes.fromNat(this.chainId || '0x1'), '0x', '0x'])
+    }
+
+    /**
+     * Fills empty optional transaction properties(gasPrice, nonce, chainId).
+     *
+     */
+    async fillTransaction() {
+        this.gasPrice = this.gasPrice === undefined ? await AbstractTransaction._klaytnCall.getGasPrice() : this.gasPrice
```

Comment:
I'm not sure it would be better, but can we execute these three functions simultaneously?

I think we can await using three promises at the end for the parallelism.

Reply:
I changed logic :)